<launch>
    <arg name="husky_namespace" default="husky" />
    <param name="use_sim_time" value="true" />


    <!-- if debug -->

    <!-- <node name="husky_inekf" pkg="husky_inekf" type="husky_estimator" output="screen" required="true" launch-prefix="gdb -ex run args"/> -->
    <node name="husky_inekf" pkg="husky_inekf" type="husky_estimator" output="screen" required="true"/>



    <!-- Run husky estimator publisher -->
    <rosparam file="$(find husky_inekf)/config/settings.yaml" command="load" />
    <!-- Initialize inEKF noise -->
    <rosparam file="$(find husky_inekf)/config/noise.yaml" command="load"/>
    <!-- Initialize inEKF prior -->
    <rosparam file="$(find husky_inekf)/config/prior.yaml" command="load"/>

    
    <node name="InEKF_path_publisher" pkg="husky_inekf" type="path_publisher_node" output="screen">
      <param name="pose_topic" value="/husky/inekf_estimation/pose"/>
      <param name="path_topic" value="/husky/inekf_estimation/path"/>
      <param name="publish_rate" value="2"/>
      <param name="pose_skip" value="1000"/>
      <!-- <param name="file_name" value="/media/curly_ssd_justin/code/minicheetah-perception/catkin_ws/src/cheetah_inekf_lcm/data/08292020_trail1_gt_2.txt"/> -->
    </node>

    <!-- <node name="InEKF_pathmap_litsener" pkg="husky_inekf" type="pathmap_litsener_node" output="screen"/> -->
    <node name="InEKF_pathodom_litsener" pkg="husky_inekf" type="pathodom_listener_node" output="screen"/>
    <node name="InEKF_gps_litsener" pkg="husky_inekf" type="gps_listener_node" output="screen"/>



    
    <!-- <node type="rviz" name="rviz" pkg="rviz">
      <param name="use_sim_time" value="true" />
    </node> -->


    <node pkg="tf" type="static_transform_publisher" name="odom_to_map" args="0 0 0 0 0 0  /map /odom 100"/>
</launch>